src/main/java/org/rrd4j/graph/RrdGraphGenerator.java

```diff
@@ -624,11 +638,17 @@ private void drawLegend() {
                     if (c instanceof LegendText) {
                         // draw with BOX
                         worker.fillRect(x, y - box, box, box, gdef.getColor(ElementsNames.frame));
-                        worker.fillRect(x + 1, y - box + 1, box - 2, box - 2,
-                                gdef.getColor(ElementsNames.canvas));
-                        worker.fillRect(x + 1, y - box + 1, box - 2, box - 2,
-                                gdef.getColor(ElementsNames.back));
-                        worker.fillRect(x + 1, y - box + 1, box - 2, box - 2, ((LegendText) c).legendColor);
+                        Paint bc = gdef.getColor(ElementsNames.back);
+                        Paint lc = ((LegendText) c).legendColor;
+                        boolean bt = bc.getTransparency() != Transparency.OPAQUE;
+                        boolean lt = lc.getTransparency() != Transparency.OPAQUE;
+                        // no use drawing unless both the two on top have some transparency
+                        if (bt && lt)
+                            worker.fillRect(x + 1, y - box + 1, box - 2, box - 2, gdef.getColor(ElementsNames.canvas));
+                        // no use drawing unless the one on top has some transparency
+                        if (lt)
```

Comment:
Can wrap all `if` in a block `{}` even one liner ?

Reply:
of course. you can do that if you're cherry-picking to match your code style.